kms/encrypt_symmetric.go

```diff
@@ -40,21 +41,41 @@ func encryptSymmetric(w io.Writer, name string, message string) error {
 	// Convert the message into bytes. Cryptographic plaintexts and
 	// ciphertexts are always byte arrays.
 	plaintext := []byte(message)
+	
+	// Optional, but recommended: Compute plaintext's CRC32C.
+	// See crc32c() function below.
+	plaintextCrc32c = crc32c(plaintext)
 
 	// Build the request.
 	req := &kmspb.EncryptRequest{
 		Name:      name,
 		Plaintext: plaintext,
+		PlaintextCrc32C: plaintextCrc32c,
 	}
 
 	// Call the API.
 	result, err := client.Encrypt(ctx, req)
 	if err != nil {
 		return fmt.Errorf("failed to encrypt: %v", err)
 	}
+	
+	// Optional, but recommended: perform integrity verification on result.
+        // For more details on ensuring E2E in-transit integrity to and from Cloud KMS visit:
+        // https://cloud.google.com/kms/docs/data-integrity-guidelines
+	if result.VerifiedPlaintextCrc32C == false {
+		return fmt.Errorf("The request sent to the server was corrupted in-transit.")
```

Comment:
> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context.

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

```suggestion
		return fmt.Errorf("Encrypt: request corrupted in-transit")
```

`Encrypt` is helpful for debugging because that's the function that was called that led to this error. Otherwise, when looking at logs, it can be unclear where the error came from.